### PR TITLE
feat: refactor project access form

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #5404 

### Description:

Screenshot 2023-12-01 at 16 22 44
Screenshot 2023-12-01 at 16 23 03
Screenshot 2023-12-01 at 16 23 14

### Reviewer actions

- [ ] I have manually tested the changes in the preview environment
- [ ] I have reviewed the code
- [ ] I understand that "request changes" will block this PR from merging

### CODE REVIEW

File: packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx [hunk 1]
```diff
@@ -1,242 +1,178 @@
...
         if (addNewMember) {
-            inviteMutation({
-                email: emailSelected,
+            const data = await inviteMutation({
+                email: formData.email,
                 role: OrganizationMemberRole.MEMBER,
             });
+            await createMutation({
+                ...formData,
+                sendEmail: false,
```
Comment (joaoviana) [inline packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx:79]: if we add a new member, shouldn't we send an email? 🤔 maybe I'm thinking about this wrong... checking render now as well

**Review by joaoviana — COMMENTED** [review]
One q! The question on props is non-blocking

**Review by joaoviana — CHANGES_REQUESTED** [review]
one q!

File: packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx (hunk 1 above)
Comment (ZeRego) [reply packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx:79]: We send an email when inviting the user to the organization, so we don't send the email inviting him specifically to this project.

**Review by joaoviana — APPROVED** [review]

File: packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx [hunk 2]
```diff
@@ -1,242 +1,178 @@
...
     return (
-        <Panel>
-            <Button
-                variant="subtle"
-                size="sm"
-                leftIcon={<MantineIcon icon={IconChevronLeft} />}
-                onClick={onBackClick}
+        <Modal
+            opened={isOpen}
+            onClose={onClose}
```
[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx:95]: ignore the above ^ that's on another `add user modal` - here for example: https://lightdash-pr-8187.onrender.com/generalSettings/userManagement

[the remark this answers is not in the document]
Comment (joaoviana) [reply packages/frontend/src/components/ProjectAccess/ProjectAccessCreation/index.tsx:95]: added an issue here https://github.com/lightdash/lightdash/issues/8208 that is unrelated to this work!

**Pull request merged**